Annotations

lsafina opened the pull request:
Skeleton for  @Invariant(expr), @StatePredicate(state, expr) and and @Pure annotations. Pre- and post-condition arguments added to @Transition.

Commit:
Route example: Added a 2s sleep to the Monitor after an add transition to avoid removing right away

Commit:
Merge branch 'master' of https://github.com/sbliudze/javabip-core

Commit:
Added the possibility to ask the executor about the current state of the component

Commit:
Modified SwitchableRoutes to output some logging information on stdout (did not manage to get the logger working)
Also added a 2 sec wait after starting to ensure the route has time to pick-up some work (not sure this actually contributes)

Commit:
Added a spontaneous port switch to the RouteOnOffMonitor, which switches a Boolean flag. When false, this flag prevents the Monitor from adding new routes (canWork guard)

Commit:
Added a new glue builder to define synchronisation of several (more than 2) ports
Added more monitor examples

Commit:
Updates for the new demo

Commit:
Minor changes:
Version change (1.6 --> 7) in pom.xml -- probably due to IntellIJ upgrade
Additional printout in Switchable roots for ease of debugging

lsafina committed:
bug-fix for Executor.getData (clazz.isAssignableFrom(methodResult.getClass())

lsafina committed:
using copyTo in datawire for deep copy

lsafina committed:
adding new annotations

lsafina committed:
added StatePredicates.java

bobismijnnaam commented:
I'm using these in the VerCors support as well :) With some small changes because of VerCors shortcomings. Some small ideas we can maybe discuss next monday:

- Should the argument name of @Invariant be "value" instead of "expr"? Then you can use the annotations like, `@Invariant("x > 2")`, instead of `@Invariant(expr = "x > 2")`.
- I think this was already discussed but putting it here for completeness: for the repeatable annotations (in particular I'm thinking of `@StatePredicates`) maybe the PR can be refactored to use `@Repeatable`? This could also be useful for other JavaBIP annotations, e.g. `@Ports`.

lsafina committed:
java expression ast for using in invariant checking

lsafina committed:
new examples

lsafina committed:
little cleaning
1) rename BIPCoordinator to GlueCoordinator,
2) add BIPException check in DataCoordinationKernel for register method, GlueCoordinatorImpl run and start methods
3) add informInternal, informSpontaneous methods skeletons

lsafina committed:
Java Expression parsing, building AST, visitors for evaluating expression

lsafina committed:
evaluating invariant in runtime, draft

lsafina committed:
moving guardsparser to the antlr v4

lsafina committed:
simplify invariant

lsafina committed:
deleted

sbliudze commented:
Hi all,

I support both proposals ;-) I had a look at repeatable annotations last
time we discussed it and, unless I misunderstood something, the change
would be really transparent for the existing JavaBIP code, since the
encompassing annotation is created automatically for the repeated ones.
Hence, I do not see any downside to using that.

Cheers,
Simon

On Fri, 1 Jul 2022 at 11:51, Bob Rubbens ***@***.***> wrote:

> I'm using these in the VerCors support as well :) With some small changes
> because of VerCors shortcomings. Some small ideas we can maybe discuss next
> monday:
>
>    - Should the argument name of @invariant <https://github.com/invariant>
>    be "value" instead of "expr"? Then you can use the annotations like, @Invariant("x
>    > 2"), instead of @Invariant(expr = "x > 2").
>    - I think this was already discussed but putting it here for
>    completeness: for the repeatable annotations (in particular I'm thinking of
>    @StatePredicates) maybe the PR can be refactored to use @Repeatable?
>    This could also be useful for other JavaBIP annotations, e.g. @Ports.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/sbliudze/javabip-core/pull/9#issuecomment-1172162418>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFKJQ2SD4Y6YLOBGEMCFMNLVR25T3ANCNFSM5PEI7B3Q>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
